From f2e5945802d40e951fa9517811a9c1cfcc70c492 Mon Sep 17 00:00:00 2001
From: "Xu,Randy" <randy.xu@intel.com>
Date: Mon, 10 Oct 2016 08:19:08 +0800
Subject: [PATCH 01/11] AndroidIA: Enable user xattr for tmpfs

It can solve the Android N live mode App crash issue. which shows
"getxattr failed: ENOTSUP"

Change-Id: I9396ea8f1230424b99256d3a804079b559c346de
Signed-off-by: Xu,Randy <randy.xu@intel.com>
---
 mm/shmem.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/mm/shmem.c b/mm/shmem.c
index 6a5973d..6b4e8f6 100644
--- a/mm/shmem.c
+++ b/mm/shmem.c
@@ -3369,6 +3369,12 @@ static int shmem_xattr_handler_set(const struct xattr_handler *handler,
 	.set = shmem_xattr_handler_set,
 };
 
+static const struct xattr_handler shmem_user_xattr_handler = {
+	.prefix = XATTR_USER_PREFIX,
+	.get = shmem_xattr_handler_get,
+	.set = shmem_xattr_handler_set,
+};
+
 static const struct xattr_handler *shmem_xattr_handlers[] = {
 #ifdef CONFIG_TMPFS_POSIX_ACL
 	&posix_acl_access_xattr_handler,
@@ -3376,6 +3382,7 @@ static int shmem_xattr_handler_set(const struct xattr_handler *handler,
 #endif
 	&shmem_security_xattr_handler,
 	&shmem_trusted_xattr_handler,
+       &shmem_user_xattr_handler,
 	NULL
 };
 
-- 
1.9.1

